core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDD.scala

```diff
@@ -95,6 +95,8 @@ private[spark] class SqlNewHadoopRDD[V: ClassTag](
   }
 
   @transient protected val jobId = new JobID(jobTrackerId, id)
+  protected val enableUnsafeRowParquetReader: Boolean =
+      sc.conf.getBoolean("spark.parquet.enableUnsafeRowRecordReader", true)
```

Comment:
we should use a sqlconf to set this, not the spark conf (spark conf is immutable once the app starts)